src/reducers/user.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { USER } from "../constants/actionTypes";
+import { type UserActions } from "../actions/user";
+import { type Profile } from "../types";
+
+export type State = {
+  auth: boolean,
+  error: string,
+  loading: boolean,
+  profile: ?Profile
+};
+const initialState: State = {
+  auth: false,
+  error: "",
+  loading: false,
+  profile: null
+};
+
+const userReducer = (
+  state: State = initialState,
+  action: UserActions
+): State => {
+  switch (action.type) {
+    case USER.REQUESTED:
+      return { auth: false, error: "", loading: true, profile: null };
+    case USER.REQUESTED_SUCCEEDED:
+      return {
+        ...state,
+        auth: true,
+        loading: false,
+        profile: action.payload
+      };
+    case USER.REQUESTED_FAILED:
+      return { ...state, error: action.payload, loading: false };
+    case USER.LOGOUT_SUCCEEDED:
```

Comment:
Logout logic should be placed inside 'auth' reducer.
If you need to delete profile from store when user logged out, just dispatch appropriate action(cleanUserReducer) from saga or somewhere else.